### ISSUE TITLE
One Step Migration Script not working

### ISSUE THREAD

**Issue by RFreiwang** [OpenMage/magento-lts#1415]
### Preconditions (*)

1. Migrating from Magento CE 1.9.2.4 to Openmage with One Step Migration Script "$ curl -fsSL https://migrate.openmage.org | sh"

### Steps to reproduce (*)

1. I cd into our development build and execute the migration script.

### Expected result (*)

1. Download the patches and apply them.

### Actual result (*)

1. Can't download or patch anything, lots of error and warnings
2. See attached log
[putty.log](https://github.com/OpenMage/magento-lts/files/5855755/putty.log)

**Comment by colinmollenhour** [comment]
Thanks for the report. The dry-run is meant to avoid breaking your install so it aborts if there are any errors before any files are overwritten.

Can you attach one of the source code files from your installation? I want to see what changes it actually contains. They appear to all be on line 20 so I'm guessing this has something to do with some patches being applied previously. 

For example:

```
error: patch failed: app/code/core/Mage/Adminhtml/Block/Catalog/Search.php:20
error: app/code/core/Mage/Adminhtml/Block/Catalog/Search.php: patch does not apply
```

**Comment by RFreiwang** [comment]
Sure, I attached 2 files. Strangely enough, both are still in the comment block on line 20. All of the other files that I looked up, too.

I changed the file format to .txt so I could upload them.

This is the app/code/core/Mage/Adminhtml/Block/Catalog/Search.php:20
[Search.txt](https://github.com/OpenMage/magento-lts/files/5865521/Search.txt)

The other one is from app/code/core/Mage/Api2/Model/Resource/Acl/Filter/Attribute/Collection.php
[Collection.txt](https://github.com/OpenMage/magento-lts/files/5865507/Collection.txt)

**Comment by sandrocantagallo** [comment]
Same problem here.

Magento version: ver. 1.9.3.1

Using commad: curl -fsSL https://migrate.openmage.org | sh

Receive many errors during path like this:

```SHELL
error: patch failed: lib/Magento/Db/Object/Interface.php:20
error: lib/Magento/Db/Object/Interface.php: patch does not apply
error: patch failed: lib/Magento/Db/Object/Table.php:20
error: lib/Magento/Db/Object/Table.php: patch does not apply
error: patch failed: lib/Magento/Db/Object/Trigger.php:20
error: lib/Magento/Db/Object/Trigger.php: patch does not apply
error: patch failed: lib/Magento/Db/Object/View.php:20
error: lib/Magento/Db/Object/View.php: patch does not apply
error: patch failed: lib/Magento/Db/Sql/Select.php:20
error: lib/Magento/Db/Sql/Select.php: patch does not apply
error: patch failed: lib/Magento/Db/Sql/Trigger.php:20
error: lib/Magento/Db/Sql/Trigger.php: patch does not apply
error: patch failed: lib/Magento/Exception.php:20
error: lib/Magento/Exception.php: patch does not apply
error: patch failed: lib/Magento/Profiler.php:20
error: lib/Magento/Profiler.php: patch does not apply
error: patch failed: lib/Magento/Profiler/Output/Csvfile.php:20
error: lib/Magento/Profiler/Output/Csvfile.php: patch does not apply
error: patch failed: lib/Magento/Profiler/Output/Firebug.php:20
error: lib/Magento/Profiler/Output/Firebug.php: patch does not apply
error: patch failed: lib/Magento/Profiler/Output/Html.php:20
error: lib/Magento/Profiler/Output/Html.php: patch does not apply
error: patch failed: lib/Magento/Profiler/OutputAbstract.php:20
error: lib/Magento/Profiler/OutputAbstract.php: patch does not apply
error: patch failed: lib/Unserialize/Parser.php:20
error: lib/Unserialize/Parser.php: patch does not apply
error: patch failed: lib/Unserialize/Reader/Arr.php:20
error: lib/Unserialize/Reader/Arr.php: patch does not apply
error: patch failed: lib/Unserialize/Reader/ArrKey.php:20
error: lib/Unserialize/Reader/ArrKey.php: patch does not apply
error: patch failed: lib/Unserialize/Reader/ArrValue.php:20
error: lib/Unserialize/Reader/ArrValue.php: patch does not apply
error: patch failed: lib/Unserialize/Reader/Bool.php:20
```

 Es: lib/Unserialize/Reader/Bool.php

In line 20 of my file i see:

` 
* @copyright  Copyright (c) 2006-2016 X.commerce, Inc. and affiliates (http://www.magento.com)
`

Patch looking for: * @copyright  Copyright (c) 2006-2018 Magento, Inc. (http://www.magento.com)

**Comment by colinmollenhour** [comment]
Thanks for the additional info, I suspect that these files were patched previously by the Magento patch scripts so the comment blocks do not match exactly the clean versions from the Magento mirror...

One option would be to download the release of the version you are running from here: https://github.com/OpenMage/magento-mirror/releases
Then extract it over your other files so that your version is identical to the clean version and then run the upgrade script.

**Comment by ADDISON74** [comment]
@colinmollenhour: I use this method but before I clean up all Magento default files. In this way I assure there is nothing left behind from previous versions, except a few custom files, themes, extensions. In the past I created a bash script for Magento 1.9.2.4 but I don't find the file right now. First it deletes all the files related to Magento then it copies from a folder the new version. Once the webserver is up and running Magento should run as expected.

**Comment by colinmollenhour** [comment]
Yes the idea is extracting the current version first would fix all of the conflicts and then running the upgrade script would apply all of the changes. Many files are removed now in OpenMage so just extracting OpenMage might work but would be suboptimal since many files that should be deleted would be left in place which could even be a security risk. Using the migration script all of the files that were removed by OpenMage would be removed without removing any user-created files like third-party extensions.

Perhaps the migration script should have a "force" mode that first extracts a clean copy?

**Comment by ADDISON74** [comment]
I downloaded the file for checking the code but it is minified. Is there a "normal" version?

**Comment by colinmollenhour** [comment]
It's not minified, the browser just renders it with no newlines. However the source is here:
https://github.com/OpenMage/migrate

**Comment by sandrocantagallo** [comment]
> Thanks for the additional info, I suspect that these files were patched previously by the Magento patch scripts so the comment blocks do not match exactly the clean versions from the Magento mirror...
> 
> One option would be to download the release of the version you are running from here: https://github.com/OpenMage/magento-mirror/releases
> Then extract it over your other files so that your version is identical to the clean version and then run the upgrade script.

I tried as suggested I downloaded from here: https://github.com/OpenMage/magento-mirror/archive/refs/tags/1.9.3.1.zip clean Magento version 1.9.3.1 the same as my project.
I have overwritten the contents of my project with the package. The site continues to work.
When I tried to run the command:

`curl -fsSL https://migrate.openmage.org | sh`

I always get the same errors such as:

```SHELL
error: patch failed: app/code/core/Mage/Admin/Helper/Block.php:20
error: app/code/core/Mage/Admin/Helper/Block.php: patch does not apply
error: patch failed: app/code/core/Mage/Admin/Helper/Data.php:20
error: app/code/core/Mage/Admin/Helper/Data.php: patch does not apply
error: patch failed: app/code/core/Mage/Admin/Helper/Rules/Fallback.php:20
error: app/code/core/Mage/Admin/Helper/Rules/Fallback.php: patch does not apply
error: patch failed: app/code/core/Mage/Admin/Helper/Variable.php:20
error: app/code/core/Mage/Admin/Helper/Variable.php: patch does not apply
error: patch failed: app/code/core/Mage/Admin/Model/Acl.php:20
error: app/code/core/Mage/Admin/Model/Acl.php: patch does not apply
error: patch failed: app/code/core/Mage/Admin/Model/Acl/Assert/Ip.php:20
error: app/code/core/Mage/Admin/Model/Acl/Assert/Ip.php: patch does not apply
error: patch failed: app/code/core/Mage/Admin/Model/Acl/Assert/Time.php:20
error: app/code/core/Mage/Admin/Model/Acl/Assert/Time.php: patch does not apply
error: patch failed: app/code/core/Mage/Admin/Model/Acl/Resource.php:20
error: app/code/core/Mage/Admin/Model/Acl/Resource.php: patch does not apply
error: patch failed: app/code/core/Mage/Admin/Model/Acl/Role.php:20
error: app/code/core/Mage/Admin/Model/Acl/Role.php: patch does not apply
error: patch failed: app/code/core/Mage/Admin/Model/Acl/Role/Generic.php:20
error: app/code/core/Mage/Admin/Model/Acl/Role/Generic.php: patch does not apply
error: patch failed: app/code/core/Mage/Admin/Model/Acl/Role/Group.php:20
error: app/code/core/Mage/Admin/Model/Acl/Role/Group.php: patch does not apply
```

If I go to check the portion of the code that gave the problem I always see the same thing:

Error

error: patch failed: app/code/core/Mage/Admin/Helper/Block.php:20

Patch

```patch
diff --git a/app/code/core/Mage/Admin/Helper/Block.php b/app/code/core/Mage/Admin/Helper/Block.php
index 83ebce5fa..dad7ab743 100644
--- a/app/code/core/Mage/Admin/Helper/Block.php
+++ b/app/code/core/Mage/Admin/Helper/Block.php
@@ -20,7 +20,7 @@
  *
  * @category    Mage
  * @package     Mage_Admin
- * @copyright  Copyright (c) 2006-2018 Magento, Inc. (http://www.magento.com)
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
  * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
```

Original File

```PHP
<?php
/**
 * Magento
 *
 * NOTICE OF LICENSE
 *
 * This source file is subject to the Open Software License (OSL 3.0)
 * that is bundled with this package in the file LICENSE.txt.
 * It is also available through the world-wide-web at this URL:
 * http://opensource.org/licenses/osl-3.0.php
 * If you did not receive a copy of the license and are unable to
 * obtain it through the world-wide-web, please send an email
 * to license@magento.com so we can send you a copy immediately.
 *
 * DISCLAIMER
 *
 * Do not edit or add to this file if you wish to upgrade Magento to newer
 * versions in the future. If you wish to customize Magento for your
 * needs please refer to http://www.magento.com for more information.
 * ---> this is line 20!!!
 * @category    Mage
 * @package     Mage_Admin
 * @copyright  Copyright (c) 2006-2016 X.commerce, Inc. and affiliates (http://www.magento.com)
 * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
 */
```

The automatic update command is definitely not working with magento versions 1.9.3.1

**Comment by sandrocantagallo** [comment]
Here is my experience of migrating a Magento portal on the OpenMage platform.

The starting project was based on Magento 1.9.3.1.

The first thing necessary since the automatic update procedure provides for the latest version was to bring Magento to version: 1.9.4.5

I didn't go for the subtlety I just downloaded a mirror:

[https://github.com/OpenMage/magento-mirror/archive/refs/tags/1.9.4.5.zip
](Magento 1.9.4.5)

And I overwrote the entire content of my project with the new source codes. I cleaned the cache and took a look at the portal and it seems that the complete purchase flow is guaranteed.

The entire original project was based on composer for installation but apparently the update on OpenMage foresees the total absence of the composer in the source project that must migrate so I have worked to delete all the files that OpenMage does not expect :

- rm .gitignore
- rm README.md
- rm composer.json

Once this is done, I launched the command provided for the migration:

`
curl -fsSL https://migrate.openmage.org | sh

`

The program responded with the following feedback:

```SHELL
Detected that you have installed Magento Commuinity Edition 1.9.4.5
Performing dry-run patch...
Dry run was successful! Applying patch...
var/magento-ce-1.9.4.5-openmage-lts-v19.4.8.patch:36602: trailing whitespace.
        ],
var/magento-ce-1.9.4.5-openmage-lts-v19.4.8.patch:37259: trailing whitespace.
        ],
var/magento-ce-1.9.4.5-openmage-lts-v19.4.8.patch:37902: trailing whitespace.
        ],
var/magento-ce-1.9.4.5-openmage-lts-v19.4.8.patch:110984: trailing whitespace.

var/magento-ce-1.9.4.5-openmage-lts-v19.4.8.patch:159713: trailing whitespace.

warning: squelched 18 whitespace errors
warning: 23 lines add whitespace errors.

All done! Please refresh your cache to complete the migration.
```

The site works correctly. 

The BUG I found is in the procedure in the Magento version: version 1.9.3.1 even if you downloaded the version of your project
